src/sentry/incidents/subscription_processor.py

```diff
@@ -550,8 +551,26 @@ def trigger_alert_threshold(
         :return:
         """
         self.trigger_alert_counts[trigger.id] += 1
+
+        # If an incident was created for this rule, trigger type, and subscription
+        # within the last 10 minutes, don't make another one
+        last_incident: Incident | None = trigger.triggered_incidents.order_by("-date_added").first()
```

Comment:
We might need to add an index on `IncidentTrigger` on `alert_rule_trigger, date_added` as well

Reply:
not sure i understand the use of an additional index here. i know it would've been helpful had we kept the original approach with `Incident.objects.filter(...)`, but since we use `trigger.triggered_incidents.order_by("-date_added").first()` to get `last_incident`, how does an additional index help us make this more efficent?

Reply:
Ahh sorry, I misread the code and thought it was sorting on `date_added` in `IncidentTrigger`, but it's in `Incident`

We might need an index on the Incident table though... `(id, date_added)`

Here's what this query will look like
```
SELECT *
FROM "sentry_incident" 
INNER JOIN "sentry_incidenttrigger" ON ("sentry_incident"."id" = "sentry_incidenttrigger"."incident_id") 
WHERE "sentry_incidenttrigger"."alert_rule_trigger_id" = 1 
ORDER BY "sentry_incident"."date_added" DESC
```

Reply:
gotcha! also, just to clarify, once the index is added, this line of code can stay the same, right?